### PR TITLE
HIVE-27113: Increasing default for hive.thrift.client.max.message.size to 2147483647 bytes

### PR DESCRIPTION
**Pull request by rtrivedi12**
**What changes were proposed in this pull request?**
Changed the default value for thrift message size to 2147483647 bytes instead of 1 GB and fixed config help message.
Also, Wrapped the config value to a max of INTEGER.MAX_VALUE.

**Why are the changes needed?**
Wide tables with huge partitions (5k+) can cross current thrift max message size of 1GB as the partition object contains column descriptors and other properties set by Impala.
Help message suggested the upper limit for max message size as 2 GB which is outside of INTEGER.MAX_VALUE range and hence caused overflow when converted to int and got updated to thrift default (100 MB)

**Does this PR introduce any user-facing change?**
No

**Is the change a dependency upgrade?**
No

**How was this patch tested?**
There is no functionality change introduced, so existing test cases should not be failing with these config value changes

### CODE REVIEW

**Comment by rtrivedi12** [comment]
@ayushtkn/ @jfsii Can you please review this?

**Review by ayushtkn — COMMENTED** [review]
Minor comments, rest looks good

**Comment by jfsii** [comment]
I agree with @ayushtkn - I think I might have suggested Math.min when the patch was originally was suggested out of caution for some reason, but I can't think of any valid reasoning today (I likely misunderstood something).

File: service/src/java/org/apache/hive/service/cli/thrift/RetryingThriftCLIServiceClient.java [hunk 1]
```diff
@@ -310,8 +310,8 @@ protected synchronized TTransport connect(HiveConf conf) throws HiveSQLException
 
     String host = conf.getVar(HiveConf.ConfVars.HIVE_SERVER2_THRIFT_BIND_HOST);
     int port = conf.getIntVar(HiveConf.ConfVars.HIVE_SERVER2_THRIFT_PORT);
-    int maxThriftMessageSize = (int) conf.getSizeVar(HiveConf.ConfVars.HIVE_THRIFT_CLIENT_MAX_MESSAGE_SIZE);
-    LOG.info("Connecting to " + host + ":" + port);
+    int maxThriftMessageSize = (int) Math.min(conf.getSizeVar(HiveConf.ConfVars.HIVE_THRIFT_CLIENT_MAX_MESSAGE_SIZE),Integer.MAX_VALUE);
```
[the remark this answers is not in the document]
Comment (rtrivedi12) [reply service/src/java/org/apache/hive/service/cli/thrift/RetryingThriftCLIServiceClient.java:313]: Agreed.. `Math.min()` is not needed here. SizeValidator enforces the check. Incorporated the change!

File: standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java [hunk 2]
```diff
@@ -610,7 +610,8 @@ public void renamePartition(String catName, String dbname, String tableName, Lis
   }
 
   private <T extends TTransport> T configureThriftMaxMessageSize(T transport) {
-    int maxThriftMessageSize = (int) MetastoreConf.getSizeVar(conf, ConfVars.THRIFT_METASTORE_CLIENT_MAX_MESSAGE_SIZE);
+    int maxThriftMessageSize = (int) Math.min(MetastoreConf.getSizeVar(
+            conf, ConfVars.THRIFT_METASTORE_CLIENT_MAX_MESSAGE_SIZE),Integer.MAX_VALUE);
```
[the remark this answers is not in the document]
Comment (rtrivedi12) [reply standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java]: Incorporated changes

File: common/src/java/org/apache/hadoop/hive/conf/HiveConf.java [hunk 3]
```diff
@@ -2928,10 +2928,10 @@ public static enum ConfVars {
     HIVE_STATS_MAX_NUM_STATS("hive.stats.max.num.stats", (long) 10000,
         "When the number of stats to be updated is huge, this value is used to control the number of \n" +
         " stats to be sent to HMS for update."),
-    HIVE_THRIFT_CLIENT_MAX_MESSAGE_SIZE("hive.thrift.client.max.message.size", "1gb",
-        new SizeValidator(-1L, true, (long) Integer.MAX_VALUE, true),
-        "Thrift client configuration for max message size. 0 or -1 will use the default defined in the Thrift " +
-        "library. The upper limit is 2147483648 bytes (or 2gb)."),
+    HIVE_THRIFT_CLIENT_MAX_MESSAGE_SIZE("hive.thrift.client.max.message.size", "2147483647b",
+            new SizeValidator(-1L, true, (long) Integer.MAX_VALUE, true),
+            "Thrift client configuration for max message size. 0 or -1 will use the default defined in the Thrift " +
```
Comment (nrg4878) [inline common/src/java/org/apache/hadoop/hive/conf/HiveConf.java:2933]: nit: is there an empty line here? Please delete if so.

File: standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java [hunk 4]
```diff
@@ -1472,9 +1472,10 @@ public enum ConfVars {
                 " corresponding service discovery servers e.g. a zookeeper. Otherwise they are " +
                 "used as URIs for remote metastore."),
     THRIFT_METASTORE_CLIENT_MAX_MESSAGE_SIZE("metastore.thrift.client.max.message.size",
-        "hive.thrift.client.max.message.size", "1gb", new SizeValidator(-1L, true, (long) Integer.MAX_VALUE, true),
-        "Thrift client configuration for max message size. 0 or -1 will use the default defined in the Thrift " +
-        "library. The upper limit is 2147483648 bytes (or 2gb)."),
+            "hive.thrift.client.max.message.size", "2147483647b",
+            new SizeValidator(-1L, true, (long) Integer.MAX_VALUE, true),
+            "Thrift client configuration for max message size. 0 or -1 will use the default defined in the Thrift " +
```
Comment (nrg4878) [inline standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java:1477]: nit: empty line?

**Review by jfsii — APPROVED** [review]
LGTM - I feel that you have addressed the prior comments. Though you will need a committer to approve.

**Review by ayushtkn — APPROVED** [review]
LGTM.

**Pull request merged**